### PR TITLE
[MM-30796] Improve automated comparison output

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR improves the output of an automated comparison by adding information about the deployment used.

It also improves the `ltctl comparison run` command by:

- Providing a new `json` output format for the comparison results.
- Providing an option to output a `zip` archive with all the artifacts (graphs, reports) included.

#### Ticket

https://mattermost.atlassian.net/browse/MM-30796

### CODE REVIEW

File: comparison/output.go [hunk 1]
```diff
@@ -67,9 +106,12 @@ func (c *Comparison) getResults(resultsCh <-chan Result) ([]Result, error) {
 		if c.config.Output.GenerateReport {
 			var buf bytes.Buffer
 			opts := report.CompareOpts{
-				GenGraph:     c.config.Output.GenerateGraphs,
-				GraphsPrefix: fmt.Sprintf("%s%d_", res.deploymentID, res.LoadTests[0].loadTestID),
+				GenGraph: c.config.Output.GenerateGraphs,
+				GraphsPrefix: fmt.Sprintf("%s_%s_%d_", res.LoadTests[0].Config.DBEngine,
```
Comment (agnivade) [inline comparison/output.go:110]: Nit: let's create the `graphsprefix` first and then assign it to the struct.

**Review by agnivade — APPROVED** [review]
Looks great. Nothing blocking.

File: cmd/ltctl/comparison.go [hunk 2]
```diff
@@ -40,6 +86,17 @@ func printResults(results []comparison.Result) error {
...
+
+func writeReports(results []comparison.Result, outPath string) error {
+	for i, res := range results {
+		if res.Report != "" {
+			filePath := filepath.Join(outPath, getReportFilename(i, res))
+			if err := ioutil.WriteFile(filePath, []byte(res.Report), 0660); err != nil {
+				return err
+			}
+		}
+	}
```
Comment (isacikgoz) [inline cmd/ltctl/comparison.go:99]: nit:
```suggestion
func writeReports(results []comparison.Result, outPath string) error {
	for i, res := range results {
		if res.Report == "" {
			continue
		}
		filePath := filepath.Join(outPath, getReportFilename(i, res))
		if err := ioutil.WriteFile(filePath, []byte(res.Report), 0660); err != nil {
			return err
		}
	}
```

**Review by isacikgoz — APPROVED** [review]
LGTM, just a nit for saving an indentation.

File: cmd/ltctl/comparison.go (hunk 2 above)
Comment (streamer45) [reply cmd/ltctl/comparison.go:99]: Nice

**Pull request merged**